### ISSUE TITLE
Does it make sense to push SNAPSHOT or preview releases to Maven Central

### ISSUE THREAD

**Issue by nicenemo** [Azure/azure-functions-java-worker#25]
Does it make sense to push SNAPSHOT versions to maven central so that  I can test the changes locally in a an easy manner withouth pulling  and building it myself?

**Comment by solvingj** [comment]
Probably not to maven central.  I would suggest the Functions team setup an account in Bintray.com, then create Nuget, Maven, NPM, and Docker repos to hold snapshots and such.  I remember the .net team had a myget repo for nightlies, but Bintray would provide one place for all their repos, and better permissions management for their developers.

**Comment by christopheranderson** [comment]
We have an AppVeyor run coming now which will produce NuGets we'll push to MyGet. No plans on moving to NuGet, yet, but it's possible we do it at GA.

AppVeyor also produces .jar files separate from the NuGet. We could at the very least publish the .jars on GitHub Releases page for folks to grab.

I would make sense to only publish well known builds to Maven Central.

**Comment by JunyiYi** [comment]
Now the only thing on Maven is the `azure-functions-java-core` package which is very stable. BUG fixes typically happen in `azure-functions-java-worker` which is never published. And it will be used by [`azure-functions-host`](https://github.com/Azure/azure-functions-host) internally. So if you want to try the latest stuffs of `azure-functions-java-worker`, you may compile it yourself and replace the built-in worker. But unfortunately you are not able to replace the built-in worker on Azure.

So in summary: no we are not going to publish SNAPSHOT version of `azure-functions-java-worker`. And there are no needs to publish any SNAPSHOT versions of `azure-functions-java-core` since it is very stable.

**Comment by solvingj** [comment]
I never saw @christopheranderson reply until now. Was there any motion on this? If not, and it does get attention in the future, I strongly suggest against github releases for this purpose (for whatever that’s worth)

**Comment by JunyiYi** [comment]
For Java runtime, we will not publish any staging runtime by ourselves.
@pragnagopa , what about the [host](https://github.com/Azure/azure-functions-host), do you have any plans to publish staging versions?

**Comment by pragnagopa** [comment]
@JunyiYi - Can you please clarify, are you looking for myget packages for azure-functions-java-worker?

**Comment by JunyiYi** [comment]
@pragnagopa , yes that's right. Customers are trying to get some staging runtime so they are able to dogfood some new features before they are published.

**Comment by pragnagopa** [comment]
As part of functions runtime release, we update all the dependent nuget packages and deploy them to public [myget feed](https://www.myget.org/gallery/azure-appservice). From now on, if there are any changes to the the java worker we can push them to myget and also update functions runtime reference. Will that work?

**Comment by JunyiYi** [comment]
@pragnagopa , That would be prefect.

**Comment by JunyiYi** [comment]
We will not publish any SNAPSHOT version to maven central. Because it will definitely break something. Always using the latest version is a bad idea.
